swap/tests/alice_punishes.rs

```diff
@@ -0,0 +1,173 @@
+use bitcoin_harness::Bitcoind;
+use futures::future::try_join;
+use libp2p::Multiaddr;
+use monero_harness::Monero;
+use rand::rngs::OsRng;
+use std::sync::Arc;
+use swap::{
+    alice, alice::swap::AliceState, bob, bob::swap::BobState, network::transport::build,
+    storage::Database, SwapAmounts, PUNISH_TIMELOCK, REFUND_TIMELOCK,
+};
+use tempfile::tempdir;
+use testcontainers::clients::Cli;
+use uuid::Uuid;
+use xmr_btc::{bitcoin, cross_curve_dleq};
+
+/// Bob locks Btc and Alice locks Xmr. Bob does not act; he fails to send Alice
+/// the encsig and fail to refund or redeem. Alice punishes.
+/// Run with RUST_MIN_STACK=10000000
+#[tokio::test]
+async fn alice_punishes_if_bob_never_acts_after_fund() {
+    use tracing_subscriber::util::SubscriberInitExt as _;
+    let _guard = tracing_subscriber::fmt()
+        .with_env_filter("swap=info,xmr_btc=info")
+        .with_ansi(false)
+        .set_default();
+
+    let alice_multiaddr: Multiaddr = "/ip4/127.0.0.1/tcp/9876"
+        .parse()
+        .expect("failed to parse Alice's address");
+
+    let cli = Cli::default();
+    let bitcoind = Bitcoind::new(&cli, "0.19.1").unwrap();
+    dbg!(&bitcoind.node_url);
+    let _ = bitcoind.init(5).await;
+
+    let btc = bitcoin::Amount::from_sat(1_000_000);
+    let btc_alice = bitcoin::Amount::ZERO;
+    let btc_bob = btc * 10;
+
+    // this xmr value matches the logic of alice::calculate_amounts i.e. btc *
+    // 10_000 * 100
+    let xmr = 1_000_000_000_000;
+    let xmr_alice = xmr * 10;
+    let xmr_bob = 0;
+
+    let alice_btc_wallet = Arc::new(
+        swap::bitcoin::Wallet::new("alice", bitcoind.node_url.clone())
+            .await
+            .unwrap(),
+    );
+    let bob_btc_wallet = Arc::new(
+        swap::bitcoin::Wallet::new("bob", bitcoind.node_url.clone())
+            .await
+            .unwrap(),
+    );
+    bitcoind
+        .mint(bob_btc_wallet.0.new_address().await.unwrap(), btc_bob)
+        .await
+        .unwrap();
+
+    let (monero, _container) =
+        Monero::new(&cli, None, vec!["alice".to_string(), "bob".to_string()])
+            .await
+            .unwrap();
+    monero
+        .init(vec![("alice", xmr_alice), ("bob", xmr_bob)])
+        .await
+        .unwrap();
+
+    let alice_xmr_wallet = Arc::new(swap::monero::Wallet(
+        monero.wallet("alice").unwrap().client(),
+    ));
+    let bob_xmr_wallet = Arc::new(swap::monero::Wallet(monero.wallet("bob").unwrap().client()));
+
+    let amounts = SwapAmounts {
+        btc,
+        xmr: xmr_btc::monero::Amount::from_piconero(xmr),
+    };
```

Comment:
I'd separate all this wallet funding step in a function that can be used by several tests.